src/Jackalope/Transport/DoctrineDBAL/CachedClient.php

```diff
@@ -476,14 +519,35 @@ public function getWeakReferences($path, $name = null)
     /**
      * {@inheritDoc}
      */
+    public function query(Query $query)
+    {
+        if (empty($this->caches['query'])) {
+            return parent::query($query);
+        }
+
+        $this->assertLoggedIn();
+
+        $cacheKey = "query: {$query->getStatement()}, {$query->getLimit()}, {$query->getOffset()}, {$query->getLanguage()}, ".$this->workspaceName;
```

Comment:
oh and anyways, i assume getStatement will expand the parameters (called bindValues) once its implemented, so there is no issue here.